New serverless pattern - SQS-EventBus using EventBridge Pipes

chaitanyagummadi opened the pull request:
*Issue #, if available:*

https://github.com/aws-samples/serverless-patterns/issues/2467

*Description of changes:*
This pattern establishes an EventBridge Pipe that retrieves messages from an Amazon Simple Queue Service (SQS) queue and sends the events to a Custom EventBus within Amazon EventBridge.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/README.md:
```diff
@@ -0,0 +1,95 @@
+# Amazon SQS queue to Amazon EventBridge custom event bus using Amazon EventBridge Pipe
+
+This pattern sends SQS messages to an Event Bus using EventBridge Pipe.
```
```suggestion
This pattern sends messages from an SQS queue to an EventBridge event bus using an EventBridge pipe.
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/README.md:
```diff
@@ -0,0 +1,95 @@
+
+This pattern sends SQS messages to an Event Bus using EventBridge Pipe.
+
+
+
```
delete additional blank lines

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/README.md:
````diff
@@ -0,0 +1,95 @@
+
+```
+
+Optional: You could create an EventBridge rule with the below event pattern on the custom bus and add relevant targets. With this setup, Message2 would be filtered out by the Pipe, and only Message1 would invoke the configured EventBridge target. Ensure to delete the EventBridge rule before the clean up proecess.
````
```suggestion
Optional: You could create an EventBridge rule with the below event pattern on the custom bus and add relevant targets. With this setup, Message2 would be filtered out by the Pipe, and only Message1 would invoke the configured EventBridge target. Ensure to delete the EventBridge rule before the clean up process.
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/README.md:
````diff
@@ -0,0 +1,95 @@
+```
+
+Optional: You could create an EventBridge rule with the below event pattern on the custom bus and add relevant targets. With this setup, Message2 would be filtered out by the Pipe, and only Message1 would invoke the configured EventBridge target. Ensure to delete the EventBridge rule before the clean up proecess.
+
+
````
delete additional empty lines

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/README.md:
````diff
@@ -0,0 +1,95 @@
+    terraform show
+    ```
+----
+Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
````
```suggestion
Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+{
+  "title": "Amazon SQS queue to Amazon EventBridge custom event bus using Amazon EventBridge Pipes",
```
The title can have at most 75 characters, please reword

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+{
+  "title": "Amazon SQS queue to Amazon EventBridge custom event bus using Amazon EventBridge Pipes",
+  "description": "This pattern creates an Amazon EventBridge Pipe to deliver messages from an Amazon SQS queue to Amazon EventBridge Event Bus.",
+  "language": "HCL",
```
```suggestion
  "language": "",
```

bfreiberg replied:
Can be removed as its the default for Terraform anyway

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+  "description": "This pattern creates an Amazon EventBridge Pipe to deliver messages from an Amazon SQS queue to Amazon EventBridge Event Bus.",
+  "language": "HCL",
+  "level": "200",
+  "framework": "terraform",
```
```suggestion
  "framework": "Terraform",
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+  },
+  "deploy": {
+    "text": [
+      "terraform init",
+       "terraform apply"
```
```suggestion
      "terraform init",
      "terraform apply"
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+    {
+      "name": "Chaitanya Gummadi",
+      "image": "https://gravatar.com/avatar/a9b30c6e727f613f3f44a44016e36998719404ea7451c750f402a21f2dd72937.jpg?size=256",
+      "bio": "Chaitanya is a Sr. Cloud Support Engineer with Amazon Web Services ( AWS) based in Texas.",
```
```suggestion
      "bio": "Chaitanya is a Senior Cloud Support Engineer with Amazon Web Services (AWS) based in Texas.",
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/main.tf:
```diff
@@ -0,0 +1,115 @@
+    }
+  }
+}
+provider "aws" {
```
Either explain to the user in the instructions how to deploy to a different region or remove this block

chaitanyagummadi replied:
Removed the region/profile block

bfreiberg commented on eventbridge-webhooks/1-stripe/template.yml:
Was this intended to be part of this pattern?

bfreiberg commented on eventbridge-webhooks/2-github/template.yaml:
Was this intended to be part of this pattern?

bfreiberg commented on eventbridge-webhooks/3-twilio/template.yaml:
Was this intended to be part of this pattern?

bfreiberg commented:
Hi, do you still plan to address the requested changes?

chaitanyagummadi commented:
> Hi, do you still plan to address the requested changes?

I have implemented the requested changes and squashed the commits to make it cleaner.

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/eventbridge-pipes-sqs-to-eventbus-terraform",
+      "templateURL": "serverless-patterns/eventbridge-pipes-sqs-to-eventbus-terraform",
+      "projectFolder": "eventbridge-pipes-sqs-to-eventbus-terraform",
+      "templateFile": "eventbridge-pipes-sqs-to-eventbus-terraform/main.tf"
```
```suggestion
      "templateFile": "main.tf"
```

bfreiberg commented on eventbridge-pipes-sqs-to-eventbus-terraform/example-pattern.json:
```diff
@@ -0,0 +1,59 @@
+  },
+  "cleanup": {
+    "text": [
+      "terraform destroy"
```
```suggestion
      "<code>terraform destroy</code>"
```

bfreiberg commented:
Looks good, thanks for your contribution. Your PR will be merged to Serverlessland.com soon.
